### PR TITLE
Adds 8 "cantrips" totally not stolen from anywhere to the spellbook

### PR DESCRIPTION
**Pull request by MrMelbert**
- Acid Touch
   - A touch spell which applies acid to whatever you touch.
   - Stronger against objects, and even stronger against walls. Costs increases with strength.
      - Can melt a table in one casts, but takes 3-4 to destroy an airlock. 
      - Can melt an iron wall and anything weaker, but not reinforced walls. 

- Mage Hand
   - It gives you telekinesis for X seconds. 
      - Yeah this is kinda lame, I wanted to make it actually do the hand thing but I figure telekinesis is more intuitive. 

- Shock Touch
   - Literally just genetics shock touch but available in magical form. 
   - Has an interaction if you get the actual genetics shock touch. 

- Illusion
   - Summons an illusion of the caster at the target location for a short time.
   - The illusion becomes less obvious the brighter the area is.
 
- Healing Touch
   - A touch spell that heals someone after a do_after. 
   - Also heals pain. 

- Soothe
   - After a short do-after, makes the target a pacifist (mechanical effect) and gives them a alert saying they have been calmed (roleplay effect). 
   - This do-after can be resisted if done to players (by pressing "resist" or attacking). 
   - If cast on AI mobs, such as Pun Pun, it will also stop them from attacking people / being hostile. 

- Finger Flame
   - The Arm Lighter Implant, in magical form. Basically just an on demand lighter with cool flavor.

- Water Control
   - "Wet a dry spot, or dry a wet spot."
   - Target turf and all adjacent either becomes wet, if they're dry, or dry, if they're wet.
   - Making dry costs nothing but mana.
   - Making wet uses water. 
      - There is an innate pool of water you have access to so you can always cast it, but this takes a while to recharge after use. 
      - You can also hold a bucket or beaker of water to draw from it, once expending your innate water.

### CODE REVIEW

File: maplestation_modules/code/modules/client/preferences/spellbook/items/spellbook_item_electrokinesis.dm [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+/datum/spellbook_item/spell/shock_touch
+	name = "Shock Touch"
+	description = "Empower your hand to deliver a shock to your target. \
+		While the zap is not powerful enough to stun or kill, it will cause them to drop their held items - or restart a stopped heart."
+	lore = "A spell which has gained notoriety for its utility in the medical field. \
+		Many wizards have found use in it, to jolt a patient out of a coma, or to restart a heart which has stopped beating. \
+		However, it is also a favorite of clowns, who use it as a less-obvious joybuzzer.\n\n\
+		Oddly, some Geneticists have found latent genes in their test subjects which allow them to use a similar ability, \
+		without the need for learning any magic or using any mana. It has yet to be determined if humankind has an innate, untapped ability \
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/client/preferences/spellbook/items/spellbook_item_electrokinesis.dm:11]: shock touch isnt human-only, so "humankind" doesn't really make too much sense

human_oid_kind, maybe?

File: maplestation_modules/code/modules/client/preferences/spellbook/items/spellbook_item_geomancy.dm [hunk 2]
```diff
@@ -0,0 +1,13 @@
+GLOBAL_LIST_INIT(spellbook_geomancy_items, generate_spellbook_items(SPELLBOOK_CATEGORY_GEOMANCY))
+
+/datum/spellbook_item/spell/acid_touch
+	name = "Acid Touch"
+	description = "Empowers your finger with a sticky acid, melting anything (or anyone) you touch."
+	lore = "A very volatile spell which has resulted in many a wizard accidentily melting off their own hands. \
+		The acid generated is comparable to that of sulfuric acid.\nThis has lead to the spell gaining wide use \
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/client/preferences/spellbook/items/spellbook_item_geomancy.dm:7]:
```suggestion
		The acid generated is comparable to that of sulfuric acid.\nThis has led to the spell gaining wide use \
```

File: maplestation_modules/code/modules/magic/story_spells/components/touch_component.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#define COMSIG_SPELL_TOUCH_CAN_HIT "spell_touch_can_hit"
+
+/datum/component/uses_mana/story_spell/touch
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/magic/story_spells/components/touch_component.dm:3]: I'd recommend adding autodoc here.

**Comment by CRITAWAKETS** [comment]
Would be interesting if Soothe gained psicodine's sanity-stabilizing ability to make it more than just RP and pacifism.

File: maplestation_modules/code/modules/client/preferences/spellbook/items/spellbook_item_healing.dm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	category = SPELLBOOK_CATEGORY_LIFE_MAGIC
+
+	our_action_typepath = /datum/action/cooldown/spell/touch/healing_touch
+
+/datum/spellbook_item/spell/soothe
+	name = "Soothe"
+	description = "Attempt to soothe a target, stopping them from feeling rage, fear, doubt, or similar emotions for a short time. \
+		This effect can be resisted by sentient targets, but also works on more simple-minded creatures."
+	lore = "A spell that is often used by by clergical figures, psychologists, or nurses to calm down those who cannot be reasoned with \
+		due to its ability to provide tranquility those in a state of panic or fear. \n\
```
Comment (siliconOpossum) [inline maplestation_modules/code/modules/client/preferences/spellbook/items/spellbook_item_healing.dm:22]:
```suggestion
		due to its ability to provide tranquility to those in a state of panic or fear. \n\
```

File: maplestation_modules/code/modules/magic/story_spells/healing_touch.dm [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+/obj/machinery/hydroponics/proc/healing_touched(datum/source, datum/action/cooldown/spell/touch/healing_touch/spell, mob/living/carbon/caster)
+	SIGNAL_HANDLER
+
+	adjust_plant_health(-1 * (spell.brute_heal + spell.burn_heal))
+	adjust_toxic(-2 * spell.tox_heal)
+	if(pestlevel > 1)
+		adjust_pestlevel(pestlevel * 0.5) // increases the amount of pests, you just healed them!
+	if(weedlevel > 2)
+		adjust_pestlevel(weedlevel * 0.5) // increases the amount of weeds, you just healed them!
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/magic/story_spells/healing_touch.dm:177]:
```suggestion
		adjust_weedlevel(weedlevel * 0.5) // increases the amount of weeds, you just healed them!
```

**Review by CRITAWAKETS — CHANGES_REQUESTED** [review]
This is probably not intentional.

**Pull request merged**